pulsar-metadata/src/test/java/org/apache/pulsar/metadata/MetadataStoreTest.java

```diff
@@ -300,4 +300,30 @@ public void notificationListeners(String provider, String url) throws Exception
         assertEquals(n.getType(), NotificationType.ChildrenChanged);
         assertEquals(n.getPath(), key1);
     }
+
+    @Test(dataProvider = "impl")
```

Comment:
Should we add Timeout for this test? the `join()` operation will wait until completed, which may block the test

Reply:
All tests have already a default timeout of 5min if not specified.